plugin/type.go

```diff
@@ -30,3 +30,13 @@ type ComposePlugin interface {
 	PostKillTask(taskInfo *mesos.TaskInfo) error
 	Shutdown(executor.ExecutorDriver) error
 }
+
+// PodStatusHook allows custom implementations to be plugged when a Pod (mesos task) status changes. Currently this is
+// designed to be executed on task status changes during LaunchTask.
+type PodStatusHook interface {
+	// Execute is invoked when task status channel has a new status
+	Execute(podStatus string, data interface{}) error
+	// BestEffort is invoked in case a Execute returned an error and is expected to return a bool to indicate
+	// if the execution needs to continue with the next available hook or not
+	BestEffort() bool
```

Comment:
Good point. Need this to determine whether execution need to proceed to next available hook. It would be better to pass this data in Execute( ) response such as (bool, error). Combining BestEffort() with Execute,  as well  would make it flexible to handle various situations -- some may be best effort while others could be not.

Reply:
Yeah, makes sense to combine this or introduce a custom error type. I will make that change.